keras_retinanet/bin/train.py

```diff
@@ -151,74 +149,47 @@ def create_generators(args):
     # create random transform generator for augmenting training data
     transform_generator = random_transform_generator(flip_x_chance=0.5)
 
-    if args.dataset_type == 'coco':
-        # import here to prevent unnecessary dependency on cocoapi
-        from ..preprocessing.coco import CocoGenerator
+    generators = None
 
-        train_generator = CocoGenerator(
-            args.coco_path,
-            'train2017',
-            transform_generator=transform_generator,
-            batch_size=args.batch_size
-        )
-
-        validation_generator = CocoGenerator(
-            args.coco_path,
-            'val2017',
-            batch_size=args.batch_size
-        )
-    elif args.dataset_type == 'pascal':
-        train_generator = PascalVocGenerator(
-            args.pascal_path,
-            'trainval',
-            transform_generator=transform_generator,
-            batch_size=args.batch_size
-        )
-
-        validation_generator = PascalVocGenerator(
-            args.pascal_path,
-            'test',
-            batch_size=args.batch_size
-        )
-    elif args.dataset_type == 'csv':
-        train_generator = CSVGenerator(
-            args.annotations,
-            args.classes,
-            transform_generator=transform_generator,
-            batch_size=args.batch_size
-        )
+    for plugin in PluginManagerSingleton.get().getAllPlugins():
+        try:
+            getattr(plugin.plugin_object, 'dataset_type')
+        except AttributeError:
+            print("Plugin ({}) does not contain a definition for dataset_type and cannot be utilised.".format(plugin.name))
+        else:
+            if plugin.plugin_object.dataset_type.lower() == args.dataset_type.lower():
+                if callable(getattr(plugin.plugin_object, 'get_generator', None)):
+                    generators = plugin.plugin_object.get_generator(args, transform_generator=transform_generator)
+                    break
+                else:
+                    print("Plugin ({}) does not contain a definition for get_generator and cannot be utilised".format(plugin.name))
```

Comment:
Same here, a warning would make sense.

I would also prefer to move the checking of plugins into the `load_plugins` function (possibly by delegating to a `check_dataset_plugin` function) and just assume they are correct here.

Reply:
Delegation of checking sounds like a good idea. Similar justification behind check args currently.